pkg/viewer/transaction.go

```diff
@@ -15,3 +15,116 @@ limitations under the License.
 */
 
 package viewer
+
+import (
+	"fmt"
+	"github.com/go-pg/pg/v10"
+	"k8s.io/klog/v2"
+
+	"github.com/bestchains/bc-explorer/pkg/models"
+)
+
+type TransArg struct {
+	//query multiple transactions
+	//params:
+	//from 			(default 0)
+	//size 			(default 10) -> last 10 transactions
+	//network 		(must)
+	//startTime
+	//endTime
+	//key
+
+	From, Size         int
+	NetworkName        string
+	StartTime, EndTime int64
+	Hash               string
+	BlockNum           uint64
+}
+
+func (ta *TransArg) ToCond() ([]string, []interface{}) {
+	params := make([]interface{}, 0)
+	cond := make([]string, 0)
+
+	if ta.NetworkName != "" {
+		cond = append(cond, ` network = ?`)
+		params = append(params, ta.NetworkName)
+	}
+	if ta.StartTime > 0 {
+		cond = append(cond, `"createdAt">=?`)
+		params = append(params, ta.StartTime)
+	}
+	if ta.EndTime > 0 {
+		cond = append(cond, `"createdAt"<=?`)
+		params = append(params, ta.EndTime)
+	}
+
+	return cond, params
+}
+
+type Transaction interface {
+	// ListTx query transactions
+	ListTx(ta TransArg) ([]models.Transaction, int64, error)
+
+	// GetTxByHash query transaction by transaction hash
+	GetTxByHash(ta TransArg) (*models.Transaction, error)
+
+	// GetTxByBlockNum query transactions by blockNumber
+	GetTxByBlockNum(ta TransArg) ([]models.Transaction, int64, error)
+}
+
+type TxHandler struct {
+	db *pg.DB
+}
+
+func NewTxHandler(db *pg.DB) Transaction {
+	return &TxHandler{db: db}
+}
+
+func (t *TxHandler) ListTx(ta TransArg) ([]models.Transaction, int64, error) {
+
+	var hd TxHandler
+	if ta.NetworkName == "" {
+		return nil, 0, fmt.Errorf("channel name can't be empty")
+	}
+
+	txs := make([]models.Transaction, 0)
+	query, params := ta.ToCond()
+	klog.Infof(" list query %s\n", query)
+
+	q := hd.db.Model(&txs)
+	for i := 0; i < len(query); i++ {
+		q = q.Where(query[i], params[i])
+	}
+
+	c, err := q.Count()
+	if err != nil {
+		return txs, 0, err
+	}
+	q = q.Order(`createdAt desc`)
+	if ta.Size != 0 {
+		q = q.Limit(ta.Size).Offset(ta.From)
+	}
+
+	if err = q.Select(); err != nil {
+		return txs, 0, err
+	}
+	return txs, int64(c), nil
+}
+
+func (t *TxHandler) GetTxByHash(ta TransArg) (*models.Transaction, error) {
+	var tx = new(models.Transaction)
+	_, err := t.db.QueryOne(&tx, `select * from transactions where "id" = ?`, ta.Hash)
+	if err != nil {
+		return nil, err
+	}
+	return tx, err
+}
+
+func (t *TxHandler) GetTxByBlockNum(ta TransArg) ([]models.Transaction, int64, error) {
+	var txs []models.Transaction
+	_, err := t.db.Query(&txs, `select * from transactions where "blockNumber" = ?`, ta.BlockNum)
+	if err != nil {
+		return nil, 0, err
+	}
```

Comment:
Search based on `blockNumber` will return multiple items of data and will require consideration of paging.

Reply:
Merged the `queryByBlockNumber` in `ListTx`, therefore it will be limited to 10 items per search by default.